src/bika/lims/content/client.py

```diff
@@ -173,29 +173,118 @@ class Client(Organisation):
     implements(IClient, IDeactivable)
 
     security = ClassSecurityInfo()
-    displayContentsTab = False
     schema = schema
-    _at_rename_after_creation = True
+    GROUP_KEY = "_client_group_id"
 
     def _renameAfterCreation(self, check_auto_id=False):
         from bika.lims.idserver import renameAfterCreation
         renameAfterCreation(self)
 
-    security.declarePublic("getContactFromUsername")
+    @property
+    def group_id(self):
+        """Client group ID
+        """
+        if not hasattr(self, self.GROUP_KEY):
+            setattr(self, self.GROUP_KEY, self.getClientID() or self.getId())
```

Comment:
The `ClientID` as the Group ID was chosen because it makes more sense to the people and allows to search for the client ID in the groups panel. The `getId` fallback was mainly because of the tests, where we do not provide the `ClientID` on sample creation.
Said that, the stored attribute handles exactly that use case where the Client ID changed afterwards.

I agree that the `getId` would be better, but having the people in mind who are using it, I decided to go with the `ClientID`.